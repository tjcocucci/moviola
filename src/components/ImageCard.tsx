import Image from 'next/image';

const ImageCard = ({
  imageUrl,
  title,
}: {
  imageUrl: string;
  title: string;
}) => {
  return (
    <div className="max-w-xs rounded overflow-hidden shadow-lg">
      <div className="relative">
        <Image
          src={imageUrl}
          alt={title}
          layout='responsive'
          width={500}
          height={300}
        />
      </div>
      <div className="px-6 py-4">
        <div className="font-bold text-xl mb-2">{title}</div>
      </div>
    </div>
  );
};

export default ImageCard;
