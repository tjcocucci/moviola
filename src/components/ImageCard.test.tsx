import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import ImageCard from './ImageCard';

const defaultProps = {
  movieObject: {
    id: 1,
    title: 'test',
    poster_path: 'test',
  },
};

describe('[ components / ImageCard ]', () => {
  describe('when component is rendered', () => {
    it('should render snapshoted ImageCard', () => {
      const { asFragment } = render(<ImageCard {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
