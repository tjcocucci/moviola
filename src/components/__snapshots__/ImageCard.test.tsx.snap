// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ components / ImageCard ] > when component is rendered > should render snapshoted ImageCard 1`] = `
<DocumentFragment>
  <div
    class="relative drop-shadow shadow-white basis-1/2 lg:basis-1/4 xl:basis-1/5 border border-gray-200 rounded-lg overflow-hidden flex flex-col items-center justify-between h-full opacity-90 hover:opacity-100 hover:scale-105 "
  >
    <div
      class="relative flex-grow-0"
    >
      <div
        class="object-cover object-center drop-shadow-2xl shadow-white"
      >
        <img
          alt="test"
          data-nimg="1"
          decoding="async"
          height="450"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500test&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
      </div>
      <div>
        <h3
          class="w-full text-m font-bold p-4"
        >
          test
        </h3>
      </div>
    </div>
    <div
      class="w-full p-4 flex flex-col justify-stretch"
    >
      <button
        class="w-full text-white font-bold py-2 px-4 rounded hover:scale-105 hover:cursor-pointer bg-blue-500 hover:bg-blue-700"
      >
        Add to List
      </button>
    </div>
  </div>
</DocumentFragment>
`;
